app/controllers/lists_controller.rb

```diff
@@ -0,0 +1,143 @@
+class ListsController < ApplicationController
+
+  before_action :set_list, except: [:new, :create, :index]
+  before_action :require_same_user, only: [:edit, :update, :flop, :add_product, :remove_product]
+  before_action :set_skroutz_connection, only: [:show, :add_product, :skus_associated_to_list, :search_product]
+  before_action :skus_associated_to_list, only: [:show, :search_product, :products_for_pagination]
+  before_action :products_for_pagination, only: [:show, :search_product]
+  
+  # Get /lists/new
+  def new; end
+  
+  # Post /lists
+  def create
+    @list = List.new(list_params)
+    @list.user = current_user
+    if @list.save
+      flash[:success] = "List was successfully created"
+      redirect_to user_path(current_user)
+    else
+      flash[:danger] = @list.errors.full_messages.first
+      redirect_to user_path(current_user)
+    end
+  end
+  
+  # Get /lists/:id/edit
+  def edit; end
+  
+  # Put /lists/:id
+  def update
+    if @list.update(list_params)
+      flash[:success] = "List name was successfully updated"
+      redirect_to user_path(current_user)
+    else
+      flash[:danger] = @list.errors.full_messages.first
+      redirect_to user_path(current_user)
+    end
+  end
+  
+  # Get /lists
+  def index
+    if params[:pagination_order].present? && params[:pagination_order]=="votes"
+      @votes_order=true
```

Comment:
You do not need this var. At view of index you can directly check if params[:pagination_order] is set to votes

Reply:
Done